features/step_definitions/command_steps.rb

```diff
@@ -216,28 +216,34 @@ def checkRestart(host, time_out)
   puts "Creating snapsnot failed..." unless $?.success?
 end
 
-When(/^I run "([^"]*)" on "([^"]*)"$/) do |cmd, target|
-  case target
-  when "server"
-    _out, $fail_code = $server.run(cmd, false)
-  when "ceos-minion"
-    _out, $fail_code = $ceos_minion.run(cmd, false)
-  when "ssh-minion"
-    _out, $fail_code = $ssh_minion.run(cmd, false)
-  when "sle-minion"
-    _out, $fail_code = $minion.run(cmd, false)
-  when "sle-client"
-    _out, $fail_code = $client.run(cmd, false)
-  when "sle-migrated-minion"
-    _out, $fail_code = $client.run(cmd, false)
-  else
-    raise "Invalid target."
-  end
+When(/^I run "([^"]*)" on "([^"]*)"$/) do |cmd, host|
+  node = get_target(host)
+  _out, $fail_code = node.run(cmd, false)
 end
 Then(/^the command should fail$/) do
    raise "Previous command must fail, but has NOT failed!" if $fail_code.zero?
 end
 
+When(/^"(.*)" exists on the filesystem of "(.*)"$/) do |file, host|
+  node = get_target(host)
+  begin
+    Timeout.timeout(DEFAULT_TIMEOUT) do
```

Comment:
No. It would probably lead to failures, as some file creations are asynchronous (for example through action chains).

What we could do is either do a function with no timeout for the synchronous cases (if any), or do a function with smaller timeout or a parametrized timeout. But none of this should be done without previous accurate analysis.